### DISCUSSION TITLE
bash: nvm: command not found

### DISCUSSION THREAD

**Discussion by JavontaeH** [coreybutler/nvm-windows#947] [Support] [answered]
Installed Nvm to directory: C:/tools/nvm along with symlink folder C:/tools/nodejs

![image](https://user-images.githubusercontent.com/61520550/225761021-1f518a59-4cb7-4989-981a-1de8bd16a3ad.png)

and setup path/enviornment variables to work properly, verified that this same installation process worked on another machine but doesn't seem to be working on mine. any ideas?

![image](https://user-images.githubusercontent.com/61520550/225760922-356eff22-b22f-400d-a60f-93110538b3a1.png)

**Comment by coreybutler** [comment, marked as answer]
These settings all look correct. However; NVM4W doesn't run in bash. It will run in a terminal or powershell, but bash is only available in the Windows Linux Subsystem.

**Reply by JavontaeH** [reply]
hmmm, I'm able to run it through git bash terminal on my local machine and also a windows terminal preview running git bash, using the same git bash terminal on a work machine is giving a command not found error as well as powershell with these settings, any other ideas?

**Reply by JavontaeH** [reply]
![image](https://user-images.githubusercontent.com/61520550/225922551-3dea85f3-6e5e-4cb4-8922-f271205a6fff.png)

**Reply by coreybutler** [reply]
The executable will display the help, but bash isn't using Windows paths. `C:/tools/nvm` and `C:/tools/nodejs` are treated as `/c/tools/nvm` and `/c/tools/nodejs`. You can see an example of this in the title bar of the shell where it shows your current path as `/c/Users/harde`. When `nvm install` or `nvm use` are run, NVM4W uses Windows paths when the shell only understands unix paths. I never had enough time to include a proper bash path translator. This is why it's not a supported shell.

https://github.com/coreybutler/nvm-windows/wiki/Common-Issues#special-terminal-shells

**Reply by JavontaeH** [reply]
This is interesting, without changing anything it seems to have randomly worked for a while, but over the weekend its not being detected once again, any ideas? I'm completely lost on this one as install seems identical to mine.

**Reply by coreybutler** [reply]
Again, bash for Windows modifies the shell in ways I've not looked into. I'm not sure why it would work sometimes and not other times, but this is why we recommend the standard Windows shells.

**Reply by JavontaeH** [reply]
oh sorry, meant to clarify that we've moved to powershell now, managed to get nvm working again but changing npm versions does not seem to work, verified ability to switch between 19.7.0 and 10.16.3 but when on 10.16.3 nvm wants to use npm v9.2.0 which is incompatible, on my system it correctly switches to npm 6.9.0. Sorry for dragging this thread out into multiple different issues but any ideas?

**Reply by JavontaeH** [reply]
trying to run npm install -g npm@6.9.0 prompted this:

`ERROR: npm v9.2.0 is known not to run on Node.js v10.16.3. You'll need to upgrade
to a newer Node.js version in order to use this version of npm. This version of
npm supports the following node versions: `^14.17.0 || ^16.13.0 || >=18.0.0`. You
can find the latest version at [https://nodejs.org/.](https://nodejs.org/) ERROR:
C:\Users\thomas.sauer\AppData\Roaming\npm\node_modules\npm\lib\utils\exit-handler.js:22
  const hasLoadedNpm = npm?.config.loaded`

**Reply by coreybutler** [reply]
NVM4W installs the version of npm associated with the version found at https://nodejs.org/dist/index.json. This means that somehow (no idea how), your system has an overridden copy of npm. Did you, by chance, copy an npm directory from one version of Node to another? This is the only way you could get a newer version of npm installed in an older version of node.

Bottom line, this comes down to an npm configuration issue outside of NVM4W. NVM4W doesn't touch npm other than the initial download/copy into the Node version's install root.

**Reply by JavontaeH** [reply]
okay replaced his old version folder with mine and everything seeeems to be working now, thank you so much for the help!
